### PR TITLE
DOC: Remove references to PyCObject

### PR DESCRIPTION
**Pull request by eric-wieser**
This has been replaced by PyCapsule, and no longer exists as of 3.2.

### CODE REVIEW

File: doc/source/reference/arrays.interface.rst [hunk 1]
```diff
@@ -315,25 +316,35 @@ largely aesthetic.  In particular:
...
-3. The tuple returned from __array_interface__['data'] used to be a
+   .. note::
+
+       Until August 2020, this said:
+
+           Now it must be a tuple whose first element is a string with "PyArrayInterface Version #" and whose
+           second element is the object exposing the array.
+
+       This design was retracted in <https://mail.python.org/pipermail/numpy-discussion/2006-June/020995.html>, and is an error to have relied
+       upon in the intermediate 14 years.
```
Comment (charris) [inline doc/source/reference/arrays.interface.rst:334]: "in" <- "it"

Comment (charris) [reply doc/source/reference/arrays.interface.rst:334]: Should also break the line.

Comment (charris) [reply doc/source/reference/arrays.interface.rst:334]: Maybe "and its continued use has been an error since that decision."

Comment (eric-wieser) [reply doc/source/reference/arrays.interface.rst:334]: I think that decision was almost immediately after this document was published. Happy to reword this however

Comment (charris) [reply doc/source/reference/arrays.interface.rst:334]: > Happy to reword this however

I found it a bit confusing, so a rewrite would help.

Comment (eric-wieser) [inline doc/source/reference/arrays.interface.rst:334]:
```suggestion
       This design was retracted in
       <https://mail.python.org/pipermail/numpy-discussion/2006-June/020995.html>,
       and it is an error to have relied upon in the intermediate 14 years.
```

Comment (eric-wieser) [inline doc/source/reference/arrays.interface.rst:334]: How's this @charris?
```suggestion
       This design was retracted almost immediately after it was proposed, in
       <https://mail.python.org/pipermail/numpy-discussion/2006-June/020995.html>.
       Despite 14 years of documentation to the contrary, at no point was it valid
       to assume that ``__array_interface__`` capsules held this tuple content.
```

Comment (eric-wieser) [reply doc/source/reference/arrays.interface.rst]: Committed my suggestion below, let me know if that's any better.

File: doc/source/reference/arrays.interface.rst [hunk 2]
```diff
@@ -315,25 +316,37 @@ largely aesthetic.  In particular:
...
+
+       Until August 2020, this said:
+
+           Now it must be a tuple whose first element is a string with "PyArrayInterface Version #" and whose
+           second element is the object exposing the array.
+
+       This design was retracted almost immediately after it was proposed, in
+       <https://mail.python.org/pipermail/numpy-discussion/2006-June/020995.html>.
+       Despite 14 years of documentation to the contrary, at no point was it valid
+       to assume that ``__array_interface__`` capsules held this tuple content.
```
Comment (eric-wieser) [inline doc/source/reference/arrays.interface.rst:336]:
```suggestion
   .. note::

       Until August 2020, this said:

           Now it must be a tuple whose first element is a string with
           "PyArrayInterface Version #" and whose second element is the object
           exposing the array.

       This design was retracted almost immediately after it was proposed, in
       <https://mail.python.org/pipermail/numpy-discussion/2006-June/020995.html>.
       Despite 14 years of documentation to the contrary, at no point was it
       valid to assume that ``__array_interface__`` capsules held this tuple
       content.
```

**Comment by eric-wieser** [comment]
@charris, are you happy with the updated wording?

File: doc/source/reference/arrays.interface.rst [hunk 3]
```diff
@@ -315,25 +316,39 @@ largely aesthetic.  In particular:
...
+           exposing the array.
+
+       This design was retracted almost immediately after it was proposed, in
+       <https://mail.python.org/pipermail/numpy-discussion/2006-June/020995.html>.
+       Despite 14 years of documentation to the contrary, at no point was it
+       valid to assume that ``__array_interface__`` capsules held this tuple
+       content.
+
+3. The tuple returned from ``__array_interface__['data']`` used to be a
    hex-string (now it is an integer or a long integer).
```
Comment (mattip) [inline doc/source/reference/arrays.interface.rst:341]:
```suggestion
3. The tuple returned from ``__array_interface__['data']`` used to be a
   hex-string (now it is an integer or a long integer).
   .. note::

       Until August 2020, this said:

           Now it must be a tuple whose first element is a string with
           "PyArrayInterface Version #" and whose second element is the object
           exposing the array.

       This design was retracted almost immediately after it was proposed, in
       <https://mail.python.org/pipermail/numpy-discussion/2006-June/020995.html>.
       Despite 14 years of documentation to the contrary, at no point was it
       valid to assume that ``__array_interface__`` capsules held this tuple
       content.

```

Comment (mattip) [reply doc/source/reference/arrays.interface.rst:341]: Ahh, right. The "this" got me confused. So maybe "Until August 2020, the description of `context` said...", or just forget it and what you have is good.

**Comment by charris** [comment]
Thanks Eric.

**Pull request merged**